app/controllers/application_controller.rb

```diff
@@ -16,6 +18,22 @@ def after_sign_in_path_for(user)
 
   private
 
+  def set_locale
+    I18n.locale = if params[:locale].to_s.to_sym.in?(I18n.available_locales)
+      params[:locale]
+    else
+      I18n.default_locale
+    end
+  end
+
+  def default_url_options(options = {})
+    options.merge({locale: resolve_locale})
+  end
+
+  def resolve_locale(locale = I18n.locale)
+    locale == I18n.default_locale ? nil : locale
+  end
```

Comment:
If this is a helper method, should it be private? Genuine question, not implying anything.

Reply:
Oh, I never thought about this.
But I believe it's good to left only actions as public in a controller?
[This](http://lifeonubuntu.com/why-your-controller-methods-should-be-private/) explains better than me.